core/src/main/java/cucumber/api/event/TestCaseEvent.java

```diff
@@ -1,4 +1,9 @@
 package cucumber.api.event;
 
 public interface TestCaseEvent {
+
+    String getFeatureUri();
```

Comment:
These two methods, instead of `TestCase getTestCase()` reduces the number of changes required

Reply:
Is there something that prevented you from using use the PickleLocation?

Reply:
Makes sense. I think that for the total ordering it may be beneficial to turn this into a PickleLocation, or if the PickleLocation is not comparable, our own Location implementation that is. It would make the TreeMap used elsewhere a bit  simpler too.

Reply:
I reviewed what was available to all of the calling methods/classes.

I went with this, as not everything had access to a `TestCase` or `PickleLocation` and as I was going to encapsulate getting the values anyway it became cleaner to inject only the URI and line number.

Reply:
It smells like primitive obsession. But maybe we can hide it later with the comparator/comparable implementation.

Reply:
You could argue that, but like I said all I needed was URI and line number to be able to locate the parent feature and sort the test case.  Hence I added these methods to the API, having the separate methods would reduce method chaining and stop the usages of the Events knowing too much about implementation detail.

How the implementation is done e.g. String + int or TestCase is an implementation detail.  All of the calls already had knowledge of the URI and line number, whereas only a few had access to something like `TestCase` - which is what we originally discussed as this is already in the `TestCaseStarted` and `TestCaseFinished` events

I did originally pass in `TestCase` but this resulted in a lot more classes have to change along with some public method signatures.